clients/src/main/java/org/apache/kafka/clients/Metadata.java

```diff
@@ -127,6 +127,15 @@ public synchronized void addTopics(String... topics) {
     }
 
     /**
+     * Add one or more topics to maintain metadata for
+     */
+    public synchronized void addTopics(Iterable<String> topics) {
+        for (String topic : topics)
+            this.topics.add(topic);
+        requestUpdate();
```

Comment:
I'd suggest to surround this command with an if-condition as below (if passing the argument as a collection):

```
if (!topics.isEmpty()) 
   requestUpdate();
```